app.rb

```diff
@@ -0,0 +1,188 @@
+      print 'Please enter a valid age: '
+      age = gets.chomp.to_i
+    end
+
+    print 'Name: '
+    name = gets.chomp.capitalize
+    while name == ''
+      print 'Please enter a name: '
+      name = gets.chomp.capitalize
+    end
+
+    print 'Has parent permission? (Y/N) '
+    permission = gets.chomp.upcase
+
+    # rubocop:disable Lint/UnreachableLoop
+    while permission
+      case permission
+      when 'Y'
+        permission = true
+      when 'N'
+        permission = false
+      else
+        print 'Please enter Y for Yes or N for No: '
+        permission = gets.chomp.upcase
+      end
+      break
+    end
+    # rubocop:enable Lint/UnreachableLoop
+
+    student = Student.new(age, nil, name, parent_permission: permission)
+    @people.push(student)
+    puts 'Person created successfully'
+    puts
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    while age < 1
+      print 'Please enter a valid age: '
+      age = gets.chomp.to_i
+    end
+
+    print 'Name: '
+    name = gets.chomp.capitalize
+    while name == ''
+      print 'Please enter a name: '
+      name = gets.chomp.capitalize
+    end
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    teacher = Teacher.new(age, specialization, name)
+    @people.push(teacher)
+    puts 'Person created successfully'
+    puts
+  end
+
+  def create_book
+    print 'Book title: '
+    title = gets.chomp.capitalize
+    while title == ''
+      print 'Book title: '
+      title = gets.chomp.capitalize
+    end
+
+    print 'Book author: '
+    author = gets.chomp.capitalize
+    while author == ''
+      print 'Book author: '
+      author = gets.chomp.capitalize
+    end
+
+    book = Book.new(title, author)
+    @books.push(book)
+    puts 'Book Created successfully'
+    puts
+  end
+
+  def create_rental
+    if @books.empty?
+      puts 'Sorry there are no books avalibales'
+      return
+    end
+    puts 'Select a book from the follwoing list by number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+    rented_book = gets.chomp.to_i
+
+    # rubocop:disable Lint/UnreachableLoop
+    while @books[rented_book].nil?
+      print 'Please select a book by index: '
+      rented_book = gets.chomp.to_i
+      break
+    end
+    # rubocop:enable Lint/UnreachableLoop
+
+    if @people.empty?
+      print 'Sorry you have to create a person first'
+      return
+    else
+      puts 'Select a person from the follwoing list by number (not id)'
+    end
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    renter = gets.chomp.to_i
+
+    # rubocop:disable Lint/UnreachableLoop
+    while @people[renter].nil?
+      print 'Please select a person by index: '
+      renter = gets.chomp.to_i
+      break
+    end
+    # rubocop:enable Lint/UnreachableLoop
```

Comment:
- [ ] Please try to avoid disabling linters as the help enforce good coding style which is better for you in the long run 👍